### PR TITLE
Issue #11531: Fixed methods for package-info file check

### PR DESCRIPTION
**Pull request by as23415672**
Fixed the issue #11531 by checking if the file name is equal to "package-info.java".

### CODE REVIEW

**Comment by as23415672** [comment]
Issue #11531: Fixed methods for package-info file check

**Comment by romani** [comment]
Make single commit

File: src/main/java/com/puppycrawl/tools/checkstyle/api/FileContents.java [hunk 1]
```diff
@@ -334,6 +335,11 @@ public Map<Integer, List<TextBlock>> getBlockComments() {
      */
     @Deprecated(since = "10.2")
     public boolean inPackageInfo() {
-        return getFileName().endsWith("package-info.java");
+        final String filePath = getFileName();
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/FileContents.java:338]: Why not `getFileName().equals("package-info.java")` ?

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/CheckUtilTest.java [hunk 2]
```diff
@@ -443,6 +443,28 @@ public void testParseClassNames() {
             .isEqualTo(expected);
     }
 
+    @Test
+    public void testIsPackageInfo() {
+        assertWithMessage("Should return true when is package info")
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/CheckUtilTest.java:448]: We do not need pure UTs methods.
All should be tested by Input files.

If you need extra testing, please create files in our resources folder to let Checkstyle run on them and skip them.

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/CheckUtilTest.java [hunk 3]
```diff
@@ -443,6 +444,24 @@ public void testParseClassNames() {
...
+        final String path = getPath("package-info.java");
+
+        assertWithMessage("Should return true when is package info")
+                .that(CheckUtil.isPackageInfo(path))
+                .isTrue();
+    }
+
+    @Test
+    public void testIsNotPackageInfo() throws IOException {
+        final String path = getPath("InputCheckUtil-package-info.java");
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/CheckUtilTest.java:458]: Please remove such tests.
Only test we allow are by Inputs and execution of whole checkstyle

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/CheckUtilTest.java (hunk 2 above)
Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/utils/CheckUtilTest.java]: Please do not resolve my comments, I will do this. Reply "done".

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/CheckUtilTest.java (hunk 3 above)
Comment (as23415672) [reply src/test/java/com/puppycrawl/tools/checkstyle/utils/CheckUtilTest.java]: Done

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/CheckUtilTest.java (hunk 2 above)
Comment (as23415672) [reply src/test/java/com/puppycrawl/tools/checkstyle/utils/CheckUtilTest.java]: Done

**Review by romani — APPROVED** [review]
ok to merge

**Review by strkkk — APPROVED** [review]

**Pull request merged**